src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,352 @@
+    remaining fields, starting with the initialization (i.e., iteration 0).
+- `ma_info::MAInfo`, structure that holds information relevant only to moving average, 
+   like the two choices of moving average widths (lambda1 and lambda2), the current moving average
+   width (lambda), a flag to indicate which moving average regime, between lambda1 and lambda2, is 
+   being used, a vector containing the values to be averaged (res_window), and an indicator of where 
+   the next value should be placed (idx).
+- `resid_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `iota_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `width_hist::Vector{Int64}`, data structure that contains the widths used in the moving average 
+   calculation at each iteration.
+- `resid_norm::Function`, a function that accepts a single vector argument and returns a
+    scalar. Used to compute the residual size.
+- `iterations::Int64`, the number of iterations of the solver.
+- `converged::Bool`, a flag to indicate whether the system has converged by some measure.
+- `true_res`, a boolean indicating if we want the true residual computed instead of approximate.
+- `dist_info::DistInfo`, a structure that stores information about the sub-Exponential distribution.
+# Constructors
+- Calling `LSLogMA()` sets `collection_rate = 1`,  `lambda1 = 1`,
+    `lambda2 = 30`, `resid_hist = Float64[]`, `iota_hist = Float64[]`, `width_hist = Int64[]`, 
+    `resid_norm = norm` (Euclidean norm), `iterations = -1`, `converged = false`, 
+    `sampler = LinSysVecRowDetermCyclic`, `max_dimension = 0`, `sigma2 = nothing`, `omega = nothing`,
+    `eta = 1`, and `true_res = false`. The user can specify their own values of lambda1, lambda2, sigma2, omega, and eta using 
+    key word arguments as inputs into the constructor.
+"""
+mutable struct LSLogMA <: LinSysSolverLog
+    collection_rate::Int64
+    ma_info::MAInfo
+    resid_hist::Vector{Float64}
+    iota_hist::Vector{Float64}
+    width_hist::Vector{Int64}
+    resid_norm::Function
+    iterations::Int64
+    converged::Bool
+    true_res::Bool
+    dist_info::DistInfo
+end
+
+LSLogMA(;
+        cr = 1, 
+        lambda1 = 1, 
+        lambda2 = 30, 
+        sigma2 = nothing, 
+        omega = nothing, 
+        eta = 1, 
+        true_res = false
+       ) = LSLogMA( cr,
+                    MAInfo(1, lambda2, 1, false, 1, zeros(lambda2)),
+                    Float64[], 
+                    Float64[], 
+                    Int64[],
+                    norm, 
+                    -1, 
+                    false,
+                    true_res,
+                    DistInfo(nothing, 0, 0, sigma2, omega, eta) 
+                  )
```

Comment:
It might be better just to have these as default constructors. Keeps things modular (but it also has downsides to doing so).

Reply:
I really think the default constructors would be bad from a usability standpoint, because if the user wanted to create this data structure they would have to remember the structures as well. A few important parameters are much easier to remember to me than the data structure and the order in which the parameters must appear.